src/api/pagination.rs

```diff
@@ -1,42 +1,15 @@
-use std::str::FromStr;
-
 use crate::utils::http;
 
-#[derive(Debug, Clone)]
-pub struct Link {
-    results: bool,
-    cursor: String,
-}
-
-#[derive(Debug, Default, Clone)]
-pub struct Pagination {
-    next: Option<Link>,
-}
-
-impl Pagination {
-    pub fn into_next_cursor(self) -> Option<String> {
-        self.next
-            .and_then(|x| if x.results { Some(x.cursor) } else { None })
-    }
-}
-
-impl FromStr for Pagination {
-    type Err = ();
-
-    fn from_str(s: &str) -> Result<Pagination, ()> {
-        let mut rv = Pagination::default();
-        for item in http::parse_link_header(s) {
-            let target = match item.get("rel") {
-                Some(&"next") => &mut rv.next,
-                _ => continue,
-            };
-
-            *target = Some(Link {
-                results: item.get("results") == Some(&"true"),
-                cursor: (*item.get("cursor").unwrap_or(&"")).to_string(),
-            });
-        }
-
-        Ok(rv)
-    }
+pub(super) fn next_cursor(value: &str) -> Option<&str> {
+    http::parse_link_header(value)
+        .iter()
+        .rev() // Reversing is necessary for backwards compatibility with a previous implementation
```

Comment:
I reversed the iterator since, in the case where there are multiple link headers with "rel", we will end up using the last one with the old implementation (since we overwrite all previous values we may have found). Do you think this reversal is necessary, or can we be sure that there will only be one "rel" header? If we aren't sure, let's just keep the rev; the comment should prevent people from being too confused.